gapic/src/main/com/google/gapid/views/TracerDialog.java

```diff
@@ -582,6 +589,56 @@ private void updateOnDeviceChange(Settings settings, DeviceCaptureInfo dev) {
         updatePerfettoConfigLabel(settings);
       }
 
+      private void runValidationCheck(DeviceCaptureInfo dev) {
+        if (dev == null) {
+          return;
+        }
+        setValidationStatus(models.devices.getValidationStatus(dev));
+        if (!models.devices.getValidationStatus(dev).passed) {
+          validationStatusLoader.startLoading();
+          validationStatusText.setText("Device is being validated");
+          rpcController.start().listen(models.devices.validateDevice(dev),
+              new UiErrorCallback<DeviceValidationResult, DeviceValidationResult, DeviceValidationResult>(validationStatusLoader, LOG) {
+            @Override
+            protected ResultOrError<DeviceValidationResult, DeviceValidationResult>
+              onRpcThread(Rpc.Result<DeviceValidationResult> response) throws RpcException, ExecutionException {
+              try {
+                return success(response.get());
+              } catch (RpcException | ExecutionException e) {
+                throttleLogRpcError(LOG, "LoadData error", e);
+                return error(null);
+              }
+            }
+
+            @Override
+            protected void onUiThreadSuccess(DeviceValidationResult result) {
+              setValidationStatus(result);
+            }
+
+            @Override
+            protected void onUiThreadError(DeviceValidationResult result) {
+              LOG.log(WARNING, "UI thread error while validating device");
+              setValidationStatus(result);
+            }
+          });
+        }
+      }
+
+      protected void setValidationStatus(DeviceValidationResult result) {
+        if (result.skipped) {
+          validationStatusLoader.updateStatus(true);
+          validationStatusLoader.stopLoading();
+          validationStatusText.setText("Validation skipped.");
+          validationStatus = true;
+        } else {
+          validationStatusLoader.updateStatus(result.passed);
+          validationStatusText.setText("Validation " + (result.passed ? "Passed." : "Failed. " + VALIDATION_FAILED_LANDING_PAGE));
+          validationStatusLoader.stopLoading();
+          validationStatus = result.passed;
+        }
+        (TraceInput.this).notifyListeners(SWT.Modify, new Event());
```

Comment:
Remove the `(TraceInput.this)` part. I know it's from my previous comment, I just wanted to point out what class to to call `notifyListeners` on.